### ISSUE TITLE
The relationship between total loss and the effect of the model.

### ISSUE THREAD

**Issue by HsLOL** [zylo117/Yet-Another-EfficientDet-Pytorch#554]
I used the data that was provided by the author@zylo117. And I trained the model with the pretrained weight (efficientdet-d0.pth).

Here is something about my loss.
> 
val epoch = 0 cls loss = 7.00165 reg loss = 0.49327 total loss = 7.49492
val epoch = 1 cls loss = 2.08872 reg loss = 0.29885 total loss = 2.38756
val epoch = 2 cls loss = 0.89503 reg loss = 0.22214 total loss = 1.11717
val epoch = 3 cls loss = 0.68907 reg loss = 0.19270 total loss = 0.88177
val epoch = 4 cls loss = 0.25602 reg loss = 0.09616 total loss = 0.35218
val epoch = 5 cls loss = 0.34037 reg loss = 0.15645 total loss = 0.49682
val epoch = 6 cls loss = 0.08738 reg loss = 0.16547 total loss = 0.25285
val epoch = 7 cls loss = 0.05665 reg loss = 0.19305 total loss = 0.24970
val epoch = 8 cls loss = 0.10043 reg loss = 0.24177 total loss = 0.34220
val epoch = 9 cls loss = 0.41436 reg loss = 0.14218 total loss = 0.55565

And here are the result pics with the different weights, which are gotten by training different epochs.

the result pic with efficientdet-d0_2_2700.pth
![image](https://user-images.githubusercontent.com/53011009/100535752-ac0c5e00-3256-11eb-9004-2b1d66456131.png)

the result pic with efficientdet-d0_7_7200.pth
![image](https://user-images.githubusercontent.com/53011009/100535770-ce05e080-3256-11eb-9715-ee0db3d0e299.png)

the result pic with efficientdet-d0_9_9000.pth
![image](https://user-images.githubusercontent.com/53011009/100535781-e83fbe80-3256-11eb-8d6e-a883304a5799.png)

Through the training loss, I think the best inference is the pic with the efficientdet-d0_7_7200.But in fact, I feel the best inference is the pic with the efficientdet-d0_9_9000.pth.

Could you please answer my question. Thanks!

**Comment by HsLOL** [comment]
Hi , I just found the **Q2. How low is low for losses** in the FAQs. And I run the coco_eval.py, here is the result.

> 
running coco-style evaluation on project shape, weights weights/efficientdet-d0.pth...
loading annotations into memory...
Done (t=0.00s)
creating index...
index created!
100%|█████████████████████████████████████████| 100/100 [00:11<00:00,  8.94it/s]
Loading and preparing results...
DONE (t=0.05s)
creating index...
index created!
BBox
Running per image evaluation...
Evaluate annotation type *bbox*
DONE (t=0.78s).
Accumulating evaluation results...
DONE (t=0.09s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.858
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.974
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.904
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.864
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.855
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.505
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.897
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.898
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.906
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.885

Process finished with exit code 0

Could you please explain the meanng of these parameters(the range of the IoU, area and the maxDets)?
or could you please send the link about these introduction about these paramenters?

Thanks!

**Comment by zylo117** [comment]
That's a part of the basic coco evaluation. You should refer to that.

**Comment by HsLOL** [comment]
Thanks, @zylo117 
I will have a look in the coco website.
